cmd/pinniped-proxy/src/main.rs

```diff
@@ -1,9 +1,40 @@
+use std::convert::Infallible;
+
+use anyhow::Result;
+use hyper::{Server, service::{make_service_fn, service_fn}};
 use structopt::StructOpt;
 
 // Ensure the root crate is aware of the child modules.
 mod cli;
+mod service;
 
-fn main() {
+#[tokio::main]
```

Comment:
The only place we actually see the lower level `tokio` library is here - a macro that converts the main function into an async function which can be run using `tokio`.